### PR TITLE
mountInto extension

### PR DESCRIPTION
**Pull request by gmkumar2005**
The `mountInto` extension is similar to `renderInto`. However, it creates a Resource that can be evaluated and safely disposed of immediately, making it useful for scenarios such as unit testing. 

On the other hand, `renderInto` is used to evaluate a resource and then keep it open indefinitely. This is typically used for mounting the application in the browser.

Example usage in Unit testing
```scala
test("renders empty elements") {
    val empty_div: Resource[IO, Element[IO]] = div("")
    empty_div.mountInto(rootElement).surround {
      IO {
        val expectedEl = document.createElement("div")
        val actual = dom.document.querySelector("#app > div")
        assert(actual != null, "querySelector returned null check if the query is correct")
        assertEquals(actual.outerHTML, expectedEl.outerHTML)
      }
    }
}
```

### CODE REVIEW

File: calico/src/main/scala/calico/syntax.scala [hunk 1]
```diff
@@ -35,3 +37,11 @@ extension [F[_], A](sigRef: SignallingRef[F, A])
 extension [E](e: E)
   inline def modify[F[_], A](a: A)(using m: Modifier[F, E, A]): Resource[F, Unit] =
     m.modify(a, e)
+
+extension [F[_]](component: Resource[F, Element[F]])
+  def mountInto(rootElement: F[Element[F]])(using Monad[F], Dom[F]): Resource[F, Element[F]] = {
```
Comment (armanbilge) [inline calico/src/main/scala/calico/syntax.scala:42]:
```suggestion
  def mountInto(rootElement: F[Element[F]])(using Monad[F], Dom[F]): Resource[F, Unit] = {
```

**Comment by gmkumar2005** [comment]
All review suggestions done.

**Comment by armanbilge** [comment]
Sorry, I'm looking again and I don't really understand how this is different from `renderInto`?

> On the other hand, `renderInto` is used to evaluate a resource and then keep it open indefinitely.

It doesn't keep it open indefinitely.

https://github.com/armanbilge/calico/blob/7703b4865f6f51ebfdb451022d65b860fc515dc9/calico/src/main/scala/calico/syntax.scala#L28-L29

**Comment by gmkumar2005** [comment]
Agree, My original comment is not accurate. The only difference is type of the first parameter.

**Comment by armanbilge** [comment]
Thanks for clarifying! I'm not sure about introducing this new method then. It's not really clear why one is "rendering" and the other is "mounting" and what the difference is. Instead I think doing something like this makes it more clear what is happening:

```scala
rootElement.toResource.flatMap(empty_div.renderInto(_))
```

Since you are in the context of a Munit test suite, maybe you want to use a [fixture](https://typelevel.org/munit-cats-effect/#suite-local-fixtures)? Then you could shorten it to something like this:

```scala
empty_div.renderInto(root())
```

**Comment by gmkumar2005** [comment]
Primary use case of this PR is unit testing. It can be solved using fixture as below. 
```scala
package basic

import calico.*
import calico.syntax.*
import calico.html.io.*
import calico.html.io.given
import cats.effect.IO
import cats.effect.Resource
import domutils.CalicoSuite
import fs2.dom.Element
import fs2.dom.Node
import munit.CatsEffectSuite
import munit.catseffect.IOFixture
import org.scalajs.dom
import org.scalajs.dom.document

class BasicSuite extends CalicoSuite {
  val mainApp: IOFixture[Node[IO]] = ResourceSuiteLocalFixture(
    "main-app",
    Resource.eval(rootElement)
  )

  override def munitFixtures = List(mainApp)

  test("renders empty elements") {
    val empty_div: Resource[IO, Element[IO]] = div("")
    empty_div.renderInto(mainApp()).surround {
      IO {
        val expectedEl = document.createElement("div")
        val actual = dom.document.querySelector("#app > div")
        assert(actual != null, "querySelector returned null check if the query is correct")
        assertEquals(actual.outerHTML, expectedEl.outerHTML)
      }
    } *> {
      val empty_span: Resource[IO, Element[IO]] = span("")
      empty_span.renderInto(mainApp()).surround {
        IO {
          val expectedEl = document.createElement("span")
          val actual = dom.document.querySelector("#app > span")
          assert(actual != null, "querySelector returned null check if the query is correct")
          assertEquals(actual.outerHTML, expectedEl.outerHTML)
        }
      }
    }
  }
}
``` 

Above code uses `renderInto` hence there is no need of `mountInto` 
Closing this PR

**Pull request closed without merging**